Synview-backend/middleware/auth_middleware.ts

```diff
@@ -0,0 +1,85 @@
+import { create, verify, Context, decode } from "../deps.ts";
+import { z } from "zod";
+import getToken from "./jwt.ts";
+import { Session } from "../deps.ts"
+/// Token Auth
+let key: CryptoKey;
+type AppState = {
+    session: Session
+}
+
+export const UserPayload = z.object({
+  ["username"]: z.string(),
+  ["role"]: z.string(),
+  ["id"]: z.number(),
+});
+
+export async function createToken(payload: any): Promise<string> {
+  key = await crypto.subtle.generateKey(
```

Comment:
This will regenerate the key each time. You can do `key ??=` to generate it only once. However, since you are accessing key from elsewhere, it may make sense to place this generating logic itself in its own function and ensure `key` is never directly accessed.